popup.js

```diff
@@ -1,193 +1,100 @@
 ﻿/*jshint esversion: 6 */
-var device;
-chrome.runtime.sendMessage({ name: 'fetchAssets' }, (assetList) => {
-  function generateTableHead(table, data) {
-    let thead = table.createTHead();
+
+const assetTable = document.querySelector('#assetTable');
```

Comment:
This is where the bulk of refactoring took place as I tried different things to figure out what was wrong. I think the result is much more reader friendly.